date_to_day.py

```diff
@@ -0,0 +1,48 @@
+from datetime import datetime
+import logging
+import os
+
+def gather_input(max_attempts=3):
+    for _ in range(max_attempts):
+        date = input("Enter a date (mm/dd/yyyy): ")
+        logger.info(f"User input: {date}")
+        try:
+            parsed = datetime.strptime(date, "%m/%d/%Y")
+            logger.info(f"Parsed date: {parsed}")
+            return parsed
+        except ValueError:
+            logger.error("Error - Date Format:", exc_info=True)
+            print("Invalid date format. Please try again.")
```

Comment:
enhancement:  Inform the user of how many attempts remain after each invalid input.
`print(f"Invalid date format. You have {max_attempts - attempt - 1} attempts left.")`

Reply:
The application uses a lot of info and error logging, but there might be opportunities to use debug for more granular details and warning for non-critical issues that should still be noted.

Reply:
fixed